app/views/questions/new.html.erb

```diff
@@ -5,18 +5,33 @@
         <div class="box">
           <h2 class="title-no-margin center">New Question</h2>
           <br>
-            <%= form_for @question do |f| %>
+            <%= form_for(@question) do |f| %>
               <div class="form-group">
-                <%= f.text_field :text, placeholder: 'Question', class: 'form-control', autofocus: true %>
+                <%= f.text_field :text, placeholder: 'Question',  class: 'form-control', autofocus: true %>
               </div>
               <div class="form-group">
-                <%= f.label :qset_id, 'Qset', {class: 'control-label'} %>
-                <%= f.collection_select(:qset_id, Qset.find(current_user.org).descendants, :id, :name, {}, {class: 'form-control'}) %>
+                <button type="button" class="btn btn-primary btn-theme btn-generate-q" data-toggle="modal" data-target="#modal-launch">
+                  Choose Qset
+                </button>
+                <% if !params[:qset].nil? %>
+                  <span class="qset-display qset-chosen">
+                    <%= @qsets.find(params[:qset]).name %>
+                  </span>
+                <% else %>
+                  <span class="qset-display no-qset-chosen">Qset Name</span>
```

Comment:
I think "Qset Name" is a bit confusing, maybe "No Qset chosen" or "Please choose a Qset for this question" or something like that?